Objects/typeobject.c

```diff
@@ -2882,6 +2888,284 @@ static const short slotoffsets[] = {
 #include "typeslots.inc"
 };
 
+static const PySlot_Offset pyslot_offsets[] = {
```

Comment:
This structure should be generated automatically. 
Did you write it by hand, or do you have a script for it?

Reply:
I write it by hand temporarily.

Reply:
petr. How about store this `pyslot_offsets` to an seperate internal seperate header?
1. It's too long~
2. more easy for script to generate.

Reply:
Yes!

It would be best to change the existing `typeslots.inc` rather than adding a new file:
- Remove `slot` from `PySlot_Offset` (the slots should be identified by their position in the array, like the current `slotoffsets`
- Change the existing `slotoffsets` to be an array of `PySlot_Offset`
- Change `typeslots.py` to generate `PySlot_Offset` structures instead of just numbers.
- Change all existing uses of `slotoffsets` to use `PySlot_Offset` (follow the two offsets, even for heap types)